src/main/java/bio/terra/cloudfunctions/utils/BigQueryUtils.java

```diff
@@ -0,0 +1,50 @@
+package bio.terra.cloudfunctions.utils;
+
+import com.google.cloud.bigquery.BigQuery;
+import com.google.cloud.bigquery.BigQueryOptions;
+import com.google.cloud.bigquery.FormatOptions;
+import com.google.cloud.bigquery.JobInfo;
+import com.google.cloud.bigquery.TableDataWriteChannel;
+import com.google.cloud.bigquery.TableId;
+import com.google.cloud.bigquery.WriteChannelConfiguration;
+import com.google.gson.JsonElement;
+import com.google.gson.JsonParser;
+import com.google.gson.JsonSyntaxException;
+import java.io.IOException;
+import java.nio.ByteBuffer;
+import java.nio.charset.StandardCharsets;
+import java.util.logging.Level;
+import java.util.logging.Logger;
+
+public class BigQueryUtils {
+  private static final Logger logger = Logger.getLogger(BigQueryUtils.class.getName());
+  /**
+   * Stream the data representing the BigQuery table rows.
+   *
+   * @param projectId the GCP project to which the BigQuery dataset resides
+   * @param dataset the name of the BigQuery dataset
+   * @param table the name of the BigQuery table to stream data to
+   * @param data the JSON data (in binary byte[] format) representing rows matching the table schema
+   */
+  public static void streamToBQ(String projectId, String dataset, String table, byte[] data)
```

Comment:
Can we assume that the BQ dataset will be living in the same project as the GCF?  If so, there's no need to pass the project id here.  If not, we need to update `BigQuery bigquery = BigQueryOptions.getDefaultInstance().getService();` so that the project id is included something like `BigQuery bigquery = BigQueryOptions.newBuilder().setProjectId(projectId).build().getService();`

Reply:
As a safeguard, we should always pass `projectId` config to the `BigQuery` client . Thanks for pointing this out.